src/scripts/main.js

```diff
@@ -1,3 +1,320 @@
+    button.innerText = 'Restart';
+    button.classList.replace('start', 'restart');
+
+    setGame();
+    isInitialSetup = false;
+  } else {
+    messageWin.classList.add('hidden');
+    won = false;
+    resetFields();
+    setGame();
+  }
+});
+
+const table = document.querySelector('.game-field');
+
+const setGame = () => {
+  for (let r = 0; r < rows; r++) {
+    const row = table.querySelectorAll('.field-row')[r];
+    const cells = row.querySelectorAll('.field-cell');
+
+    for (let c = 0; c < columns; c++) {
+      const cell = cells[c];
+
+      cell.setAttribute('id', `row_${r}-column_${c}`);
+
+      const number = board[r][c];
+
+      updateCell(cell, number);
+    }
+  }
+  setCell();
+  setCell();
+};
+
+const hasEmptyCell = () => {
+  for (let r = 0; r < rows; r++) {
+    for (let c = 0; c < columns; c++) {
+      if (board[r][c] === 0) {
+        return true;
+      }
+    }
+  }
+
+  return false;
+};
+
+const loseGame = () => {
+  if (hasEmptyCell()) {
+    return false;
+  }
+
+  for (let r = 0; r < rows; r++) {
+    for (let c = 0; c < rows; c++) {
+      if (board[r][c] === board[r][c + 1]) {
+        return false;
+      }
+    }
+  }
+
+  for (let r = 0; r < rows - 1; r++) {
+    for (let c = 0; c < rows; c++) {
+      if (board[r][c] === board[r + 1][c]) {
+        return false;
+      }
+    }
+  }
+
+  return true;
+};
+
+const resetFields = () => {
+  for (let r = 0; r < rows; r++) {
+    for (let c = 0; c < columns; c++) {
+      board[r][c] = 0;
+    }
+  }
+};
+
+const clearBoard = () => {
+  for (let r = 0; r < rows; r++) {
+    for (let c = 0; c < columns; c++) {
+      const cell = document.getElementById(`row_${r}-column_${c}`);
+
+      cell.innerText = '';
+      cell.classList.value = '';
+      cell.classList.add('field-cell');
+    }
+  }
+  scoreCount = 0;
+  score.innerHTML = scoreCount;
+};
+
+const setCell = () => {
+  if (!hasEmptyCell()) {
+    return;
+  }
+
+  let found = false;
+  const number = Math.random() < 0.8 ? 2 : 4;
+
+  while (!found) {
+    const row = Math.floor(Math.random() * rows);
+    const column = Math.floor(Math.random() * columns);
+
+    if (board[row][column] === 0) {
+      board[row][column] = 2;
+
+      const cell = document.getElementById(`row_${row}-column_${column}`);
+
+      cell.innerText = `${number}`;
+      cell.classList.add(`field-cell--${number}`);
+      found = true;
+    }
+  }
+
+  if (loseGame()) {
+    messageLose.classList.remove('hidden');
+
+    button.addEventListener('click', () => {
+      messageLose.classList.add('hidden');
+
+      clearBoard();
+      setGame();
+    });
+  }
+};
+
+const updateCell = (cell, number) => {
+  cell.innerText = '';
+  cell.classList.value = '';
+  cell.classList.add('field-cell');
+
+  if (number > 0) {
+    cell.innerText = number;
+
+    if (number < 4096) {
+      cell.classList.add(`field-cell--${number}`);
+    } else {
+      cell.classList.add(`field-cell--8192`);
+    }
+  }
+
+  if (number === 2048) {
```

Comment:
```suggestion
  if (number === 2048) {
```

Move magic numbers like 2048 to constant